### PR TITLE
Fixes parallel being set to 1 when True

### PR DESCRIPTION
**Pull request by dxenes1**
Previously we used to check if `Parallel=True` with `isinstance()` in the `get_cutout()` function. However this returns True when compared to `int` since `bool` is a subclass of `int` and thus sets the multiprocessing Pool to only use 1 CPU instead of all CPUs (expected behavior). Therefore I rewrote some of the code so we get the expected behavior from setting `Parallel=True`. 

https://stackoverflow.com/questions/37888620/comparing-boolean-and-int-using-isinstance

### CODE REVIEW

File: intern/service/boss/v1/volume.py [hunk 1]
```diff
@@ -251,7 +251,13 @@ def get_cutout(
             ), dtype=resource.datatype)
 
             if parallel:
-                pool = multiprocessing.Pool(processes=parallel if isinstance(parallel, int) and parallel > 0 else multiprocessing.cpu_count())
+                if type(parallel) == bool:
```
Comment (movestill) [inline intern/service/boss/v1/volume.py:254]: Let's remove the type annotation from the function or make it `Union[bool, int]` so it's not a lie. Also update the docstring to indicate that an int is also a valid type.

**Review by movestill — CHANGES_REQUESTED** [review]

**Review by j6k4m8 — APPROVED** [review]
Approved with @movestill's comments :)

**Review by j6k4m8 — APPROVED** [review]

File: intern/service/boss/v1/volume.py (hunk 1 above)
Comment (movestill) [reply intern/service/boss/v1/volume.py:255]: Still need to also hit the type annotation in the function def, itself.

Comment (dxenes1) [reply intern/service/boss/v1/volume.py:255]: Ok removed the type annotation in the v1 volume.py. By the function def did you mean the wrapper function in `service/volume.py`?

Comment (movestill) [reply intern/service/boss/v1/volume.py:255]: I was referring to the change you just made.

**Review by movestill — APPROVED** [review]

**Review by j6k4m8 — APPROVED** [review]

**Pull request merged**